lib/traversing.js

```diff
@@ -9,30 +9,34 @@
 function traverse(node, opt_onEnter, opt_onLeave) {
   if (opt_onEnter) opt_onEnter(node);
 
-  const childNodes = _collectChildNodes(node);
-  childNodes.forEach(function(childNode) {
-    traverse(childNode, opt_onEnter, opt_onLeave);
+  const childNodeInfo = _collectChildNodeInfo(node);
+  childNodeInfo.forEach(function([childNode, propName, parentType]) {
+    traverse(childNode, opt_onEnter ? (node, pn, pt) => {
+      opt_onEnter(node, pn || propName, pt || parentType);
```

Comment:
probably should be named `parentPropName` instead of `propName`.

Reply:
Done.